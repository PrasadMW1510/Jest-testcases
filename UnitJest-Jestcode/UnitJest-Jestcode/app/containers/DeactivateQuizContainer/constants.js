/*
 *
 * DeactivateQuizContainer constants
 *
 */
export const DEACTIVATE_QUIZ_LIST = 'app/DeactivateQuizContainer/deactivate_quiz_list';
export const DEFAULT_ACTION = 'app/DeactivateQuizContainer/DEFAULT_ACTION';
export const DEACTIVATE_QUIZ_LIST_FAILURE =
  'app/DeactivateQuizContainer/deactivate-quiz-list-failure';
