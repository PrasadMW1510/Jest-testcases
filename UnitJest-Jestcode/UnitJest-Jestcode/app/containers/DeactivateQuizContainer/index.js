/**
 *
 * DeactivateQuizContainer
 *
 */

export { default } from './DeactivateQuizContainer';
