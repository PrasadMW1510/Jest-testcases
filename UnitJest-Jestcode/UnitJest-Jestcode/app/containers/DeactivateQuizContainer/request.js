import API, { getBaseUrlWithoutSlms } from 'utils/request';

export const postDeactivateQuizRequest = (sessionId, data) =>
  API.post(`${getBaseUrlWithoutSlms()}/src/SrcQuizManager`, data, {
    params: {
      sid: sessionId,
      command: 'set_quiz_activation',
    },
  }).then(() => {});
