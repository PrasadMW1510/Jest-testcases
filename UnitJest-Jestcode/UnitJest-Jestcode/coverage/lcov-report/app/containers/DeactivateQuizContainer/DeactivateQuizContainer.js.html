<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/containers/DeactivateQuizContainer/DeactivateQuizContainer.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">app/containers/DeactivateQuizContainer</a> DeactivateQuizContainer.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/74</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/39</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/27</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/**<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span>
 *
 * DeactivateQuizContainer
 *
 */
&nbsp;
import React <span class="cstat-no" title="statement not covered" >from 'react';<span class="cstat-no" title="statement not covered" ></span></span>
import PropTypes <span class="cstat-no" title="statement not covered" >from 'prop-types';<span class="cstat-no" title="statement not covered" ></span></span>
import DeactivateQuiz <span class="cstat-no" title="statement not covered" >from 'components/DeactivateQuiz';<span class="cstat-no" title="statement not covered" ></span></span>
import injectSaga <span class="cstat-no" title="statement not covered" >from 'utils/injectSaga';<span class="cstat-no" title="statement not covered" ></span></span>
&nbsp;
import PopupModal <span class="cstat-no" title="statement not covered" >from 'components/ActivateQuiz/PopupModal';<span class="cstat-no" title="statement not covered" ></span></span>
import { connect }<span class="cstat-no" title="statement not covered" > from 'react-redux';</span>
import { createS<span class="cstat-no" title="statement not covered" >tructuredSelector } </span>from 'reselect';
import { comp<span class="cstat-no" title="statement not covered" >ose } from 'redux</span>';
&nbsp;
import makeSelect<span class="cstat-no" title="statement not covered" >SearchResultsContainer from 'containers/SearchResultsCo</span>ntainer/selectors'<span class="cstat-no" title="statement not covered" >;</span>
&nbsp;
import { getSearch<span class="cstat-no" title="statement not covered" >edResults } from 'containers/BookQuizContainer/sel</span>ectors';
import { hideMo<span class="cstat-no" title="statement not covered" >dal, showMessageLogModal } from 'containers/Mo</span>dalController/actions';
import { deactiv<span class="cstat-no" title="statement not covered" >ateQuizRequest } from</span> './actions';
import saga <span class="cstat-no" title="statement not covered" >from './saga';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span>
&nbsp;
export class DeactivateQuizContainer extends React.Component {
  deactivateQuiz = () =&gt; {
    const { selectedItems } = this.props.searchResultsContainer;<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span></span></span></span>
    th<span class="cstat-no" title="statement not covered" >is.props.deactivateQuizRequest(selectedItems);</span>
&nbsp;
    if<span class="cstat-no" title="statement not covered" > (selectedItems.length === 0) {</span>
      th<span class="cstat-no" title="statement not covered" >is.props.deactivateQuizRequest(false);</span>
      th<span class="cstat-no" title="statement not covered" >is.props.showMessageLogModal({</span>
        payloadData: &lt;PopupModal /&gt;,
      });
    }
  };<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span>
  render() {
    const { hideModal:<span class="cstat-no" title="statement not covered" > hideModalProp } = th</span>is.props;
&nbsp;
    re<span class="cstat-no" title="statement not covered" >turn (</span>
      &lt;div&gt;
        &lt;DeactivateQuiz
          isOpen
          handleCancel={hideModalProp}
          deactivateQuiz={this.deactivateQuiz}
          {...this.props}
        /&gt;
      &lt;/div&gt;
    );
  }<span class="cstat-no" title="statement not covered" ></span>
}
&nbsp;
<span class="cstat-no" title="statement not covered" >DeactivateQuizContainer.propTypes = {</span>
  deactivateQuizRequest: PropTypes.func,
  hideModal: PropTypes.func,
  searchResultsContainer: PropTypes.object,
  showMessageLogModal: PropTypes.func,
};
&nbsp;
const mapStateToProps <span class="cstat-no" title="statement not covered" >= createStructuredSelector({</span>
  searchedBookResults: getSearchedResults(),
  searchResultsContainer: makeSelectSearchResultsContainer(),
});
const withConnect <span class="cstat-no" title="statement not covered" >= connect(mapStateToProps, {</span>
  hideModal,
  deactivateQuizRequest,
  showMessageLogModal,
});
const withSaga <span class="cstat-no" title="statement not covered" >= injectSaga({ key: 'deactivateQuizContainer', saga });<span class="cstat-no" title="statement not covered" ></span></span>
&nbsp;
export default compose(withSaga, withConnect)(DeactivateQuizContainer);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jul 21 2018 17:06:47 GMT+0530 (India Standard Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
