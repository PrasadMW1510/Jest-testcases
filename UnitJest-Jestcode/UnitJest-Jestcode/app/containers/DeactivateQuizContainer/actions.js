/*
 *
 * DeactivateQuizContainer actions
 *
 */

import * as Constants from './constants';

export function deactivateQuizRequest(tableData, searchOpts) {
  return {
    type: Constants.DEACTIVATE_QUIZ_LIST,
    tableData,
    searchOpts,
  };
}

export function getPostDeactivateRequestFailure(error) {
  return {
    type: Constants.DEACTIVATE_QUIZ_LIST_FAILURE,
    error,
  };
}

export function defaultAction() {
  return {
    type: Constants.DEFAULT_ACTION,
  };
}
