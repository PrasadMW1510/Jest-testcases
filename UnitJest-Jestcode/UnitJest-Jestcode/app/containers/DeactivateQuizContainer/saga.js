import { makeSelectedbookresults } from 'containers/SearchResultsContainer/actions';
import { hideModal } from 'containers/ModalController/actions';
import { all, call, select, takeLatest, put } from 'redux-saga/effects';
import * as Constants from './constants';
import { makeSelectProfileSessionId } from '../App/selectors';
import * as Request from './request';
import * as Actions from './actions';

export function* postDeactivateQuizRequest(action) {
  try {
    const sessionId = yield select(makeSelectProfileSessionId());
    const tableData = action.tableData;
    // const rootElm = {
    //   SetQuizActivationReq: {
    //     ActiveState: false,
    //     Quizzes: { QuizID: tableData.map(id => id.QuizID[0]) },
    //   },
    // };
    let rootElm = '<SetQuizActivationReq><ActiveState>false</ActiveState><Quizzes>';
    tableData.map(id => {
      rootElm += `<QuizId>${id.QuizID[0]}</QuizId>`;
      return rootElm;
    });
    rootElm += '</Quizzes></SetQuizActivationReq>';
    yield call(Request.postDeactivateQuizRequest, sessionId, rootElm);
    const mappedVal = tableData.map(val => {
      const value = val;
      value.QuizActive = ['false'];
      return val;
    });
    yield put(makeSelectedbookresults(mappedVal));
    yield put(hideModal());
  } catch (err) {
    yield put(Actions.getPostDeactivateRequestFailure(err));
    yield put(hideModal());
  }
}

export default function* defaultSaga() {
  yield all([takeLatest(Constants.DEACTIVATE_QUIZ_LIST, postDeactivateQuizRequest)]);
}
