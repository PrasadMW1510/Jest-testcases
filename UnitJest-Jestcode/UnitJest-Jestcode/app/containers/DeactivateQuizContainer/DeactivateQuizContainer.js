/**
 *
 * DeactivateQuizContainer
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import DeactivateQuiz from 'components/DeactivateQuiz';
import injectSaga from 'utils/injectSaga';

import PopupModal from 'components/ActivateQuiz/PopupModal';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import makeSelectSearchResultsContainer from 'containers/SearchResultsContainer/selectors';

import { getSearchedResults } from 'containers/BookQuizContainer/selectors';
import { hideModal, showMessageLogModal } from 'containers/ModalController/actions';
import { deactivateQuizRequest } from './actions';
import saga from './saga';

export class DeactivateQuizContainer extends React.Component {
  deactivateQuiz = () => {
    const { selectedItems } = this.props.searchResultsContainer;
    this.props.deactivateQuizRequest(selectedItems);

    if (selectedItems.length === 0) {
      this.props.deactivateQuizRequest(false);
      this.props.showMessageLogModal({
        payloadData: <PopupModal />,
      });
    }
  };
  render() {
    const { hideModal: hideModalProp } = this.props;

    return (
      <div>
        <DeactivateQuiz
          isOpen
          handleCancel={hideModalProp}
          deactivateQuiz={this.deactivateQuiz}
          {...this.props}
        />
      </div>
    );
  }
}

DeactivateQuizContainer.propTypes = {
  deactivateQuizRequest: PropTypes.func,
  hideModal: PropTypes.func,
  searchResultsContainer: PropTypes.object,
  showMessageLogModal: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  searchedBookResults: getSearchedResults(),
  searchResultsContainer: makeSelectSearchResultsContainer(),
});
const withConnect = connect(mapStateToProps, {
  hideModal,
  deactivateQuizRequest,
  showMessageLogModal,
});
const withSaga = injectSaga({ key: 'deactivateQuizContainer', saga });

export default compose(withSaga, withConnect)(DeactivateQuizContainer);
