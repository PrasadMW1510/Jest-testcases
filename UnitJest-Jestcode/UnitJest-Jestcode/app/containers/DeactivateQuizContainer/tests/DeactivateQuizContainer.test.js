import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { DeactivateQuizContainer } from '../DeactivateQuizContainer';

describe('<DeactivateQuizContainer />', () => {
  let wrapper = null;
  let props = null;
  let props1 = null;
  let wrapper1 = null;
  beforeEach(() => {
    props = {
      deactivateQuizRequest: jest.fn(),
      hideModal: jest.fn(),
      searchResultsContainer: {
        selectedItems: [],
      },
      showMessageLogModal: jest.fn(),
    };
    wrapper = shallow(<DeactivateQuizContainer {...props} />);
  });
  props1 = {
    deactivateQuizRequest: jest.fn(),
    hideModal: jest.fn(),
    handleCancel: jest.fn(),
    searchResultsContainer: {
      selectedItems: ['iii'],
    },
    showMessageLogModal: jest.fn(),
  };
  wrapper1 = shallow(<DeactivateQuizContainer {...props1} />);
  it('should render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should call deactivateQuiz   ', () => {
    wrapper.instance().deactivateQuiz();
    expect(wrapper.instance().props.searchresultsselectedData).toMatchSnapshot();
  });
  it('should call deactivateQuiz else ', () => {
    wrapper1.instance().deactivateQuiz();
    expect(shallowToJson(wrapper1)).toMatchSnapshot();
  });
});
