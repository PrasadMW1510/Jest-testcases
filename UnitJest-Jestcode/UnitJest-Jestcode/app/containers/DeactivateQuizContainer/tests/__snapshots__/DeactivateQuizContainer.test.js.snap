// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeactivateQuizContainer /> should call deactivateQuiz    1`] = `undefined`;

exports[`<DeactivateQuizContainer /> should call deactivateQuiz else  1`] = `
<div>
  <DeactivateQuiz
    deactivateQuiz={[Function]}
    deactivateQuizRequest={[Function]}
    handleCancel={[Function]}
    hideModal={[Function]}
    isOpen={true}
    searchResultsContainer={
      Object {
        "selectedItems": Array [
          "iii",
        ],
      }
    }
    showMessageLogModal={[Function]}
  />
</div>
`;

exports[`<DeactivateQuizContainer /> should render correctly 1`] = `
<div>
  <DeactivateQuiz
    deactivateQuiz={[Function]}
    deactivateQuizRequest={[Function]}
    handleCancel={[Function]}
    hideModal={[Function]}
    isOpen={true}
    searchResultsContainer={
      Object {
        "selectedItems": Array [],
      }
    }
    showMessageLogModal={[Function]}
  />
</div>
`;
